examples/energy-management-app/energy-management-common/include/ChargingTargetsMemMgr.h

```diff
@@ -0,0 +1,153 @@
+ * ChargingTargetsMemMgr::PrepareDaySchedule must be called so this object knows which day schedule it is tracking.
+ * This will free any previous memory allocated for the day schedule in this object.
+ *
+ * There are then three usage cases:
+ *
+ * 1. When loading the Target from persistent storage. In this scenario, it is not known upfront
+ *    how many chargingTargets are associated with this day schedule so ChargingTargetsMemMgr::AddChargingTarget()
+ *    needs to be called as each individual chargingTarget is loaded from persistent data.
+ *
+ *    Once the chargingTargets for the day schedule have been loaded, ChargingTargetsMemMgr::AllocAndCopy() is
+ *    called to allocate the memory to store the chargingTargets and the chargingTargets are copied.
+ *
+ * 2. When updating a Target and a day schedule is unaffected, the chargingTargets associated with
+ *    day schedule need copying. The following should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::List<const Structs::ChargingTargetStruct::Type> & chargingTargets)
+ *
+ * 3. When in SetTargets, a new list of chargingTargets needs to be added to a day schedule, the following
+ *    should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::DecodableList<Structs::ChargingTargetStruct::DecodableType> &
+ * chargingTargets)
+ *
+ * Having allocated and copied the chargingTargets accordingly, they can be added to a
+ * DataModel::List<const Structs::ChargingTargetStruct::Type as follows:
+ *
+ * chargingTargetsList = DataModel::List<Structs::ChargingTargetStruct::Type>(ChargingTargetsMemMgr::GetChargingTargets(),
+ * ChargingTargetsMemMgr::GetNumDailyChargingTargets());
+ *
+ * All memory allocated by this object is released When the ChargingTargetsMemMgr destructor is called.
+ *
+ */
+
+class ChargingTargetsMemMgr
+{
+public:
+    ChargingTargetsMemMgr();
+    ~ChargingTargetsMemMgr();
+
+    /**
+     * @brief This method prepares a new day schedule. Subsequent calls to GetChargingTargets
+     *        and the AllocAndCopy methods below will reference this day schedule.
+     *
+     * @param chargingTargetSchedulesIdx  - The new day schedule index
+     */
+    void PrepareDaySchedule(uint16_t chargingTargetSchedulesIdx);
+
+    /**
+     * @brief Called as each individual chargingTarget is loaded from persistent data.
+     *        When loading the Target from persistent storage, it is not known upfront
+     *        how many chargingTargets are associated with this day schedule so
+     *        ChargingTargetsMemMgr::AddChargingTarget() needs to be called as each individual
+     *        chargingTarget is loaded from persistent data.
+     *
+     * @param chargingTarget  - The chargingTarget that will be added into the current day schedule
+     */
+    void AddChargingTarget(const EnergyEvse::Structs::ChargingTargetStruct::Type & chargingTarget);
```

Comment:
you have a limit on how large your charging targets are (kEvseTargetsMaxTargetsPerDay it seems). This method returns void so it cannot report a failure. How is failure handled? documentation on the method should explain that.

Reply:
I have added some documentation.